src/charm.py

```diff
@@ -588,6 +591,14 @@ def _on_relation_departed(self, event: RelationDepartedEvent) -> None:
 
         self._update_hosts(event)
 
+    def _on_storage_attached(self, event: StorageAttachedEvent) -> None:
+        """Handler for `storage_attached` event.
+
+        This should handle fixing the permissions for the data dir.
+        """
+        safe_exec(f"chmod -R 770 {Config.MONGODB_COMMON_PATH}".split())
+        safe_exec(f"chown -R {Config.SNAP_USER}:root  {Config.MONGODB_COMMON_PATH}".split())
```

Comment:
When we are attaching the storage, we need to fix the permissions on it.
On one side, we fixed the snap so it does not recreate and give permissions if the directories are already existing.
On this side, we grant the right permissions to the already existing directories so the snap will not break because of permissions error. If we don't fix those permissions, there will be an installation error when adding a unit with an already existing storage, as the snap will not be allowed to do anything with the existing files, not even reading the configuration file to update it as we do.
This is inspired by the work made on Kafka:
 * https://github.com/canonical/kafka-operator/pull/195
 * https://github.com/canonical/charmed-kafka-snap/pull/40